### ISSUE TITLE
Memory leak on RPi when updating texture of an Image from the camera via openCV

### ISSUE THREAD

**Issue by openedhardware** [kivy/kivy#5171]
- Python: 2.7.10
- OS: Windows 8
- Kivy: 1.10

I am trying to show live video with openCV and texture of Image widget.

Here is my app.py file:
```
class VideoPreviewWidget(BoxLayout):

    capture = None
    event_take_video = None
    texture = None

    def __init__(self, **kwargs):
        super(VideoPreviewWidget, self).__init__(**kwargs)
        self.capture = cv2.VideoCapture(0)
        self.capture.set(cv2.CAP_PROP_FRAME_WIDTH, 640)
        self.capture.set(cv2.CAP_PROP_FRAME_HEIGHT, 480)

    def on_btn_start(self):
        if self.event_take_video is None:
            # Call `self.take_video` 30 times per sec
            self.event_take_video = Clock.schedule_interval(self.take_video, 1.0 / 30.0)
        elif not self.event_take_video.is_triggered:
            self.event_take_video()

    def on_btn_stop(self):
        if self.event_take_video.is_triggered:
            self.event_take_video.cancel()

    def frame_to_buf(self, frame):
        if frame is None:
            return
        buf1 = cv2.flip(frame, 0)
        if buf1 is not None:
            buf = buf1.tostring()
            self.texture = Texture.create(size=(640, 480))
            self.texture.blit_buffer(buf, colorfmt='bgr', bufferfmt='ubyte')
            return True

    def take_video(self, *args):
        """
        Capture video frame and update image widget
        :param args:    
        :return:
        """
        try:
            ret, frame = self.capture.read()
            if self.frame_to_buf(frame=frame):
                self.ids.img_left.texture = self.texture
            else:
                self.ids.img_left.source = cur_dir + '/img/bad_camera.png'
        except:
            self.ids.img_left.source = cur_dir + '/img/bad_camera.png'

    def on_close(self):
        self.capture.release()


class VideoPreviewApp(App):

    def build(self):
        return VideoPreviewWidget()

    def on_stop(self):
        self.root.on_close()
        super(VideoPreviewApp, self).on_stop()


if __name__ == '__main__':
    VideoPreviewApp().run()
```

And here is my .kv file:

```
<VideoPreviewWidget>:
    Image:
        id: img_left
        source: "img/bad_camera.png"
        allow_stretch: True
        size_hint: None, None
        size: 640, 480
    BoxLayout:
        orientation: 'vertical'
        size_hint_x: None
        width: 160
        spacing: 40
        padding: 20, 0, 20, 0
        Widget:
        Button:
            text: 'START'
            on_release: root.on_btn_start()
            size_hint_y: None
            height: 50
        Button:
            text: 'STOP'
            on_release: root.on_btn_stop()
            size_hint_y: None
            height: 50
        Button:
            text: 'EXIT'
            on_release: app.stop()
            size_hint_y: None
            height: 50
        Widget:
```

Pretty simple...

When I execute this on Windows machine, it works well without increasements of memory usage.
But if I execute on Raspberry Pi, memory usage rises 5% per sec and it locked up due to the leak image.

Not sure what is wrong...

**Comment by openedhardware** [comment]
This is the repository of this app: https://github.com/openedhardware/video_preview

Feel free to try and add a comment!

**Comment by bionoid** [comment]
Maybe fixed by #5664, please re-test if possible
